### PR TITLE
Improve tunnel identity notifications

### PR DESCRIPTION
**Pull request by joestringer**
These two improvements were previously part of #10951 for visibility and debugging, but a different approach to the bugfix was taken. This PR just splits out the notifications changes.

/cc @jrfastab already reviewed these and gave thumbs up.

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
Hit #11029

File: bpf/bpf_overlay.c [hunk 1]
```diff
@@ -89,7 +88,7 @@ static __always_inline int handle_ipv6(struct __ctx_buff *ctx,
 		ctx_change_type(ctx, PACKET_HOST);
 		return CTX_ACT_OK;
 	} else {
-		key.tunnel_id = get_identity(ctx);
+		*identity = get_identity(ctx);
```
Comment (qmonnet) [inline bpf/bpf_overlay.c]: Is this line necessary at all with the change above? If I understand correctly we have:
``` c
if (decrypted) {
	*identity = key.tunnel_id = get_identity(ctx);
} else { ... }

cilium_dbg(...);

// No change to *identity, or decrypted, or ctx->mark

if (!decrypted) { ... }
else { // if (decrypted)
	*identity = get_identity(ctx);
}
```
So `*identity` is set twice to the same value?

File: bpf/bpf_overlay.c [hunk 2]
```diff
@@ -202,7 +200,7 @@ static __always_inline int handle_ipv4(struct __ctx_buff *ctx, __u32 *identity)
...
```
Comment (qmonnet) [inline bpf/bpf_overlay.c]: Same remark.

**Review by qmonnet — CHANGES_REQUESTED** [review]

File: bpf/bpf_overlay.c (hunk 1 above)
Comment (joestringer) [reply bpf/bpf_overlay.c]: Perhaps not. IMO it's a little easier to read if the `if/else` conditions initialize the `*identity` in the same spot. I can take a look whether that's possible to do without duplicating this line.

Comment (qmonnet) [reply bpf/bpf_overlay.c]: Not sure if we understand each other. The two `if` are consecutive, and I meant that if `decrypted` is `true`, we execute twice `*identity = get_identity(ctx);` successively, with no change to `*identity` or to the result from `get_identity(ctx);` in the meantime. Isn't the second assignment identical to the first, and thus useless?

Comment (joestringer) [reply bpf/bpf_overlay.c]: I got half way through my reply to this in the morning before github broke..

Summary, you're right, I read the code wrong when I first saw your review. Fixed it up now. Thanks for the review and the patience :-)

**Comment by joestringer** [comment]
test-me-please

File: bpf/bpf_overlay.c (hunk 1 above)
Comment (qmonnet) [reply bpf/bpf_overlay.c]: Sure, no problem :)

**Review by qmonnet — APPROVED** [review]

**Comment by qmonnet** [comment]
test-gke

**Pull request merged**